### PR TITLE
Upgrade to the latest operator-sdk

### PR DESCRIPTION
**Pull request by mprahl**
operator-sdk v1.x now wraps Kubebuilder for its scaffolding and code
generation. This changes the path of `./cmd/manager/main.go` to
`.main.go` and `./pkg/controller` to `./controllers`.

Manifest generation is now handled by kubebuilder. This is handled by
the `generate-operator-yaml` Make target. This means that any RBAC
changes must be done using kubebuilder markers and not manual edits to
the manifests in `./deploy`.

The largest changes stem from updating the underlying Kubernetes Go
modules as some APIs changed or were removed.

It's also important to note that operator-sdk is no longer a Go module
dependency, so some functionality was copied over, mostly unmodified.

Resolves:
https://github.com/open-cluster-management/backlog/issues/15550

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -148,10 +181,13 @@ func main() {
 		os.Exit(1)
 	}
 
+	// This lease is not related to leader election. This is to report the status of the controller
+	// for observability reasons.
```
Comment (JustinKuli) [inline main.go:185]: What observability reasons?

I think this comment is important, it is a confusing point. I just want to make sure we're describing the "why" correctly for ourselves later. When I hear "obsesrvability" here I might think of the observability component in ACM. Is that what this is for? Or I thought it was something to do with addon framework?

File: tool/operator-sdk-port.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+func isRunModeLocal() bool {
+	return os.Getenv(ForceRunModeEnv) == string(LocalRunMode)
+}
+
+// GetWatchNamespace returns the Namespace the operator should be watching for changes
+func GetWatchNamespace() (string, error) {
+	// WatchNamespaceEnvVar is the constant for env variable WATCH_NAMESPACE
+	// which specifies the Namespace to watch.
+	// An empty value means the operator is running with cluster scope.
+	var watchNamespaceEnvVar = "WATCH_NAMESPACE"
```
Comment (JustinKuli) [inline tool/operator-sdk-port.go:41]: Can this be a package `const` ?

File: .github/workflows/kind.yml [hunk 3]
```diff
@@ -53,6 +53,11 @@ jobs:
         make fmt
         git diff --exit-code
 
+    - name: Verify deploy/operator.yaml
+      run: |
+        make generate-operator-yaml
+        git diff --exit-code
```
Comment (JustinKuli) [inline .github/workflows/kind.yml:59]: Good idea!

**Review by JustinKuli — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (mprahl) [reply main.go]: Thanks for calling me out on this. I added a better description.

**Pull request merged**